### PR TITLE
Upgrade to vanilla v1.8.0

### LINKED ISSUE #5: Write a basic 404 page for docs
Write a 404 page that:

- Is pretty
- Mentions search
- Mentions the Discourse forum

### LINKED ISSUE #15: Upgrade to latest version of Vanilla Framework

### PR DESCRIPTION
**Pull request by nottrobin**
- Use vanilla-framework `v1.8.0` instead of `v0.1.0`
- Squeeze content to `64.875rem`, like the current docs.snapcraft.io
- Copy some styling overrides and JavaScript for docs layouts from docs.vanillaframework.io (`_vf-overrides.scss`)
- Copy wholesale from snapcraft.io:
  - `_patterns_social-icons.scss`
  - `_snapcraft_footer.scss`
  - `_snapcraft_p-navigation.scss`
- Add https://highlightjs.org for syntax highlighting
- Add some custom styling:
  - For the navigation format as it comes from the forum (which uses `<h3>`s) 
  - For `blockquote`s, the spacing didn't look quite right
  - For `pre` and `code` so they play nicely with highlightjs
  - For the huge `.emoji` images that come from Discourse
- Rip out the old vf 0.1.0-based styles
- Write some JavaScript for adding `is-active` to the selected navigation item

Fixes #15
Fixes #5 

QA
--

`./run`, and check http://0.0.0.0:8029/ looks good and works well, even on small screens.

### CODE REVIEW

File: templates/base.html [hunk 1]
```diff
@@ -1,126 +1,162 @@
...
-        </header>
+          toggle.addEventListener('click', function(e) {
+            toggle.classList.toggle('p-icon--menu');
+            toggle.classList.toggle('p-icon--close');
+            sidebarContent.classList.toggle('u-hide--small');
+          });
+
+          // Add active links to active nav item
+          document.querySelectorAll('.p-sidebar-nav ul a').forEach(function(anchor) {
+            if (anchor.pathname == location.pathname) {
```
Comment (Caleb-Ellis) [inline templates/base.html]: This isn't working for anchors that have a redirect. I assume this is something that would have to be fixed in discourse?

**Review by Caleb-Ellis — COMMENTED** [review]
- Getting an error when trying to run it: `File to import not found or unreadable: hljs-default`

File: templates/base.html (hunk 1 above)
Comment (nottrobin) [reply templates/base.html]: Yes, it would have to be fixed in Discourse.

**Review by Caleb-Ellis — APPROVED** [review]
Looks really good!

**Pull request merged**